invenio_records_rest/errors.py

```diff
@@ -29,15 +29,169 @@
 from invenio_rest.errors import RESTException
 
 
+#
+# Search
+#
 class MaxResultWindowRESTError(RESTException):
     """Maximum number of results passed."""
 
     code = 400
-    description = "Maximum number of results have been reached."
+
+    def __init__(self, max):
+        """Initialize exception."""
+        self.description = \
+            'Maximum number of results have been reached ({0}).'.format(max)
 
 
+#
+# Query
+#
 class InvalidQueryRESTError(RESTException):
     """Invalid query syntax."""
 
     code = 400
-    description = "Invalid query syntax."
+    description = 'Invalid query syntax.'
+
+
+#
+# CiteProc
+#
+class StyleNotFoundRESTError(RESTException):
+    """No such style."""
+
+    code = 400
+
+    def __init__(self, style):
+        """Initialize exception."""
+        self.description = \
+            '[CiteProc]: Style \'{0}\'could not be found.'.format(style)
+
+
+#
+# PID
+#
+class PIDDoesNotExistRESTError(RESTException):
+    """Non-existent PID."""
+
+    code = 404
+    description = 'PID does not exist.'
+
+
+class PIDUnregisteredRESTError(RESTException):
+    """Unregistered PID."""
+
+    code = 404
+    description = 'PID is not registered.'
+
+
+class PIDDeletedRESTError(RESTException):
+    """Deleted PID."""
+
+    code = 410
+    description = 'PID has been deleted.'
+
+
+class PIDMissingObjectRESTError(RESTException):
+    """PID missing object."""
+
+    code = 500
+
+    def __init__(self, pid):
+        """Initialize exception."""
+        self.description = 'No object assigned to {0}.'.format(pid)
+
+
+class PIDRedirectedRESTError(RESTException):
+    """Invalid redirect for destination."""
+
+    code = 500
+
+    def __init__(self, pid_type):
+        """Initialize exception."""
+        self.description = 'Invalid redirect - pid_type \'{0}\' ' \
```

Comment:
Besides, please don't use this:

```
      'this is \'something\` isn't it?' 
```
     
but rather one of these:

```
      'this is "something" isn't it?' 
      "this is 'something' isn't it?" 
```

see [PEP-8](https://www.python.org/dev/peps/pep-0008/):

> In Python, single-quoted strings and double-quoted strings are the same. This PEP does not make a recommendation for this. Pick a rule and stick to it. When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string. It improves readability.